functions/ensure.pp

```diff
@@ -3,7 +3,7 @@
 # @return [String]
 function stdlib::ensure(
   Variant[Boolean, Enum['present', 'absent']] $ensure,
-  Enum['directory', 'link', 'mounted', 'service', 'file', 'package'] $resource,
+  Enum['present', 'directory', 'link', 'mounted', 'service', 'file', 'package'] $resource = 'present',
```

Comment:
I don't think there is a resource type named `present`. If you want to make it optional, then perhaps it should be made `undef` and caught as an `undef` in the case statement.
```suggestion
  Optional[Enum['present', 'directory', 'link', 'mounted', 'service', 'file', 'package']] $resource = undef,
```